File: swift/test/TestDriver/factory.swift
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

import Foundation
import TestCommon
import IceProperties
import IceProxy
import IceDefaultValue
import IceEnums
import IceStream

let tests: [String: () -> TestHelper] = [
    "Ice.properties.Client": { IceProperties.Client() },
    "Ice.proxy.Client": { IceProxy.Client() },
    "Ice.defaultValue.Client": { IceDefaultValue.Client() },
    "Ice.stream.Client": { IceStream.Client() },
    "Ice.enums.Client": { IceEnums.Client() },
    "Ice.enums.Server": { IceEnums.Server() }
]

class TestFactory {
    public static func create(name: String) throws -> TestCommon.TestHelper {
        if let test = tests[name] {
            return test()
        } else {
            throw NSError(domain: "Test not found \(name)", code: 1024)
        }
    }
}

File: cpp/demo/Ice/pickle/Factory.cpp
// **********************************************************************
//
// Copyright (c) 2001
// MutableRealms, Inc.
// Huntsville, AL, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <Ice/Ice.h>
#include <Factory.h>
#include <Pickle.h>

using namespace std;

Ice::ObjectPtr
Factory::create(const string& id)
{
    if (id == "::Persistent")
	return new Persistent;

    assert(false);
    return 0;
}
